sw/device/lib/testing/hmac_testutils.c

```diff
@@ -56,70 +56,77 @@ const dif_hmac_digest_t kHmacRefExpectedDigest = {
         },
 };
 
-void hmac_testutils_check_message_length(const dif_hmac_t *hmac,
-                                         uint64_t expected_sent_bits) {
+status_t hmac_testutils_check_message_length(const dif_hmac_t *hmac,
+                                             uint64_t expected_sent_bits) {
   uint64_t sent_bits;
-  CHECK_DIF_OK(dif_hmac_get_message_length(hmac, &sent_bits));
+  TRY(dif_hmac_get_message_length(hmac, &sent_bits));
 
   // 64bit formatting is not supported, so split into hi and lo hex 32bit
   // values. These should appear as 64bit hex values in the debug output.
-  CHECK(expected_sent_bits == sent_bits,
-        "Message length mismatch. "
-        "Expected 0x%08x%08x bits but got 0x%08x%08x bits.",
-        (uint32_t)(expected_sent_bits >> 32), (uint32_t)expected_sent_bits,
-        (uint32_t)(sent_bits >> 32), (uint32_t)sent_bits);
+  TRY_CHECK(expected_sent_bits == sent_bits,
+            "Message length mismatch. "
+            "Expected 0x%08x%08x bits but got 0x%08x%08x bits.",
+            (uint32_t)(expected_sent_bits >> 32), (uint32_t)expected_sent_bits,
+            (uint32_t)(sent_bits >> 32), (uint32_t)sent_bits);
+
+  return OK_STATUS();
 }
 
 /**
  * Checks whether the HMAC FIFO is empty.
  */
 static bool check_fifo_empty(const dif_hmac_t *hmac) {
   uint32_t fifo_depth;
-  CHECK_DIF_OK(dif_hmac_fifo_count_entries(hmac, &fifo_depth));
-  return fifo_depth == 0;
+  dif_result_t res = dif_hmac_fifo_count_entries(hmac, &fifo_depth);
+  return res == kDifOk || fifo_depth == 0;
 }
 
-void hmac_testutils_fifo_empty_polled(const dif_hmac_t *hmac) {
-  IBEX_SPIN_FOR(check_fifo_empty(hmac), HMAC_TESTUTILS_FIFO_EMPTY_USEC);
+status_t hmac_testutils_fifo_empty_polled(const dif_hmac_t *hmac) {
+  IBEX_TRY_SPIN_FOR(check_fifo_empty(hmac), HMAC_TESTUTILS_FIFO_EMPTY_USEC);
+  return OK_STATUS();
 }
 
-static bool check_finished(const dif_hmac_t *hmac,
-                           dif_hmac_digest_t *digest_out) {
+static status_t check_finished(const dif_hmac_t *hmac,
+                               dif_hmac_digest_t *digest_out) {
   dif_result_t res = dif_hmac_finish(hmac, digest_out);
-  CHECK(res == kDifOk || res == kDifUnavailable, "HMAC error = %d", res);
+  TRY_CHECK(res == kDifOk || res == kDifUnavailable, "HMAC error = %d", res);
 
-  return res == kDifOk;
+  return INTO_STATUS(res);
```

Comment:
Lets keep `INTO_STATUS` as a "private" macro within `status.h`.

Based on how this function is used, I think it would be fine for it to return a `bool` just as it did before.

Reply:
@timothytrippel the reason this function exist is to be used as an argument in the `IBEX_SPIN_FOR` macro. So I'll keep it.

Reply:
Oh right, I think @cfrantz is right then, since this is a private (static) function anyway, just keep return type of bool. We just care of about the public API of the testutils.

Reply:
Function delete.